pkg/utils/baremetal/bootstrap.go

```diff
@@ -0,0 +1,100 @@
+package baremetal
+
+import (
+	"context"
+	"time"
+
+	baremetalhost "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	clientwatch "k8s.io/client-go/tools/watch"
+
+	"github.com/openshift/installer/pkg/types/baremetal"
+)
+
+// WaitForBaremetalBootstrapControlPlane will watch baremetalhost resources on the bootstrap
+// and wait for the control plane to finish provisioning.
+func WaitForBaremetalBootstrapControlPlane(ctx context.Context, config *rest.Config) error {
+	timeout := 30 * time.Minute
```

Comment:
Side note: someone will definitely ask us to make this configurable.

Reply:
I like that.  Moved to `waitForBootstrapConfigMap`.